rest-api-guidelines/quality/evolution/naming-conventions.md

```diff
@@ -13,6 +13,14 @@ Every identifier **MUST** be in American English and written in `lowercase`. An
 
 Every URI **MUST** follow the General Rules except for the `camelCase` rule. Instead, a hyphen \(`-`\) **SHOULD** be used to delimit combined words \(kebab-case\). Besides, a URI **MUST NOT** end with a trailing slash \(`/`\).
 
+Plural nouns SHOULD be used in the URI where appropriate to identify collections of data resources (e.g. ``/orders``, ``/products``).
+
+An individual resource in a collection of resources MAY exist directly beneath the collection URI.  (e.g. ``/orders/{order_id}``).
+
+Sub-resource collections MAY exist directly beneath an individual resource. This should convey a relationship to another collection of resources (e.g. ``/orders/{order_id}/items``).
+
+Sub-resource individual resources MAY exist, but should be avoided in favor of top-level resources. (e.g. ``/orders/{order_id}/items/{item_id}``).
```

Comment:
Per my comment above, I do not think this is a good recommendation even as a rule of thumb. If you want to offer an that "lists an item by its id on a particular order" This is the best way to express it, even if the items ids are unique across orders...

Reply:
im not sure if i got this. can you rephrase please?

Reply:
@jerzyn lets discuss it over a short call :) It might be more effective than here.

Reply:
Hi @zdne, @jerzyn I think I can agree with you both :)

I agree with @jerzyn as I think it is always good to explicitly express hard relations between root resources and sub-resources as orders and lineItem. Following with this example, if a lineItem does not have a meaning/lifecycle outside the context of an order, the recommendation should be to not expose that sub-resource with a root path and always use the root resource to navigate to it (via links or embedded attribute). That's btw inline with the Aggregate's approach in DDD and transactional boundaries. For instance, Can I update a lineItem without affecting the total amount of an order? If there is a hard relation URLs SHOULD make it explicit. 

However I do also agree with @zdne on making it a rule of thumb or not as it be confusing when creating compositions that aggregate resource in a way that is valid for a specific use case but don't necessarily reflect with exactness the relation between resources.

Reply:
> Following with this example, if a lineItem does not have a meaning/lifecycle outside the context of an order, the recommendation should be to not expose that sub-resource with a root path and always use the root resource to navigate to it. 

I like this as a rule of a thumb.

Reply:
> If there is a hard relation URLs SHOULD make it explicit.

Something is fighting in me about this – the URLs should really be the identifiers in the first place – one should not guess the relations types from URLs, yet many ppl are still doing it. Frankly, If I would see a Hypermedia API that has URLs in the form of UUIDs (e.g. `/00AA1FF11D2310` and `/112AAAA00003`) I would be fine with it ...